go/app/main.go

```diff
@@ -33,24 +55,136 @@ func addItem(c echo.Context) error {
+
+	fileModel := strings.Split(file.Filename, ".")
+	fileName := fileModel[0]
+	extension := fileModel[1]
+	fileName_hash := sha_256([]byte(fileName))
+
+	// 保存用のファイルを作成する
+	dst, err := os.Create("./images/"+fmt.Sprintf("%s.%s",string(fileName_hash),extension))
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+	defer dst.Close()
+
+	// アップロードされたファイルの内容を保存用のファイルにコピーする
+	_, err = io.Copy(dst, src)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+	c.Logger().Infof("アップロード成功!")
+	// ここまで画像
+	imageName := fileName_hash + "." + extension
+	//新しい商品（ジャンル）の読み取り
+	category := c.FormValue("category")
+	//新しい商品を構造体に
+	newItem := Item{Name: name, Category: category, Imagename: imageName}
+	//新しい商品を商品一覧配列に追加
+	itemsWrapper.Items = append(itemsWrapper.Items, newItem)
+
+	//商品一覧配列をjson化
+	ans, err := json.Marshal(itemsWrapper)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+	os.WriteFile("./app/items.json", []byte(ans), 0664)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+
 	return c.JSON(http.StatusOK, res)
 }
 
+func showItem (c echo.Context) error {
+	//jsonファイルの読み込み
+	raw, err := os.ReadFile("./app/items.json")
+	//エラーハンドリング
+	if err != nil {
+        return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+    }
+	var itemsWrapper ItemsWrapper
+	if err := json.Unmarshal(raw, &itemsWrapper); err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+	return c.JSON(http.StatusOK, itemsWrapper)
+}
+
 func getImg(c echo.Context) error {
 	// Create image path
 	imgPath := path.Join(ImgDir, c.Param("imageFilename"))
+	fmt.Println(imgPath)
 
 	if !strings.HasSuffix(imgPath, ".jpg") {
 		res := Response{Message: "Image path does not end with .jpg"}
 		return c.JSON(http.StatusBadRequest, res)
 	}
 	if _, err := os.Stat(imgPath); err != nil {
+		c.Logger().SetLevel(log.DEBUG)
 		c.Logger().Debugf("Image not found: %s", imgPath)
 		imgPath = path.Join(ImgDir, "default.jpg")
 	}
+
 	return c.File(imgPath)
 }
 
+func getItems(c echo.Context) error {
+	item_id := c.Param("item_id")
+
+	//jsonファイルの読み込み
+	raw, err := os.ReadFile("./app/items.json")
+	//エラーハンドリング
+	if err != nil {
+        return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+    }
+	//jsonに対応した構造体のインスタンスを生成
+	var itemsWrapper ItemsWrapper
+	//jsonをデコード
+	if err := json.Unmarshal(raw, &itemsWrapper); err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+
+	items := itemsWrapper.Items
+	index, err := strconv.Atoi(item_id)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, Response{Message: err.Error()})
+	}
+	if len(items) < index {
+		message := fmt.Sprintf("")
+		res := Response{Message: message}
+		return c.JSON(http.StatusOK, res)
```

Comment:
[nits] ここはエラーを定義するのでもいいかもしれませんね。
```suggestion
		message := fmt.Sprintf("invalid item_id")
		res := Response{Message: message}
		return c.JSON(http.StatusBadRequest, res)
```